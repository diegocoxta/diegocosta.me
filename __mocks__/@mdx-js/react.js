module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
