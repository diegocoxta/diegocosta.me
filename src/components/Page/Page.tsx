import React from 'react';
import { createGlobalStyle, ThemeContext } from 'styled-components';

import Header from '~/components/Header';
import Footer from '~/components/Footer';

import { useTheme } from '~/hooks/useTheme';
import { dark, light } from '~/themes';

type CustomTheme = typeof dark;

const GlobalStyle = createGlobalStyle<{ theme: CustomTheme }>`
  body {
    background: ${({ theme }) => theme.backgroundColor};
    font-family: 'Raleway', sans-serif;
    color: #fff;
    margin: 0;
    padding: 0;
  }
`;

export interface PageProps {
  children: React.ReactNode;
}

export default function Page(props: PageProps): React.ReactElement {
  const [theme, themeToggler] = useTheme();

  const themeMode = theme === 'light' ? light : dark;

  return (
    <ThemeContext.Provider value={{ ...themeMode, theme, themeToggler }}>
      <GlobalStyle />
      <Header />
      <main>{props.children}</main>
      <Footer />
    </ThemeContext.Provider>
  );
}
