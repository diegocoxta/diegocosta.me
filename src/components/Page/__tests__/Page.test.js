import React from 'react';
import { render } from '@testing-library/react';

import Page from '../Page';

jest.mock('../../Header', () => () => <p>Header</p>);
jest.mock('../../Footer', () => () => <p>Footer</p>);

describe('<Page />', () => {
  it('renders properly', () => {
    const { baseElement, queryByText } = render(<Page>John Doe</Page>);
    expect(queryByText('John Doe')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
