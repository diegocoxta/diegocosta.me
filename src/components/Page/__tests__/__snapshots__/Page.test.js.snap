// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render properly 1`] = `
<body>
  <div>
    <p>
      Header
    </p>
    <main>
      John Doe
    </main>
    <p>
      Footer
    </p>
  </div>
</body>
`;
